File: lib/src/main/java/com/androidessence/lib/FadeInSpan.java
package com.androidessence.lib;

import android.text.TextPaint;
import android.text.style.CharacterStyle;
import android.text.style.UpdateAppearance;

/**
 * Created by Raghunandan on 28-11-2016.
 */

public class FadeInSpan extends CharacterStyle implements UpdateAppearance {

    private int alpha;

    public int getAlpha() {
        return alpha;
    }

    public void setAlpha(int alpha) {
        this.alpha = alpha;
    }

    public FadeInSpan() {

    }

    @Override
    public void updateDrawState(TextPaint paint) {
        paint.setAlpha(alpha);

    }
}

File: lib/src/main/java/com/androidessence/lib/FadeInSpan.kt
package com.androidessence.lib

import android.text.TextPaint
import android.text.style.CharacterStyle
import android.text.style.UpdateAppearance

/**
 * Span that allows text to fade in.
 *
 * Created by Raghunandan on 28-11-2016.
 */
class FadeInSpan : CharacterStyle(), UpdateAppearance {

    var alpha: Int = 0

    override fun updateDrawState(paint: TextPaint) {
        paint.alpha = alpha
    }
}